subsys/bluetooth/host/Kconfig

```diff
@@ -12,11 +12,13 @@ config BT_LONG_WQ
 
 if BT_LONG_WQ
 config BT_LONG_WQ_STACK_SIZE
-	# Hidden: Long workqueue stack size. Should be derived from system
-	# requirements.
-	int
-	default BT_HCI_ECC_STACK_SIZE if BT_TINYCRYPT_ECC
+	int "Long workqueue stack size"
```

Comment:
I don't think we can have a prompt on this variable. Variables with a prompt are loaded from `.config` if that's available, and then the defaults do not take effect. So if a user compiles without `BT_GATT_CACHING` and then enables it without deleting `.config`, the value of `BT_LONG_WQ_STACK_SIZE` would not be updated even if it's too small. (Yes, kconfig is weird and difficult to use. I don't like it either.)

Reply:
On the other hand, I think that user needs possibility to configure size of this stack in some cases (e.g. to tailor the stack to save memory or to increase it on some architectures - we may not oversee everything and I would prefer not to make stack very big by default).

Reply:
This seems like limitation of Kconfig, so maybe we could assume that users are aware of it. Or maybe you have a proposition of an alternative solution here?

Reply:
I think `CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE` and `CONFIG_BT_RX_STACK_SIZE` are broken.

Menuconfig exists, so we must support interactive config setup. And we must handle the user experimenting with turning `BT_GATT_CACHING` on or off, etc., in a sane way. Conditional defaults for prompts completely break this workflow.

If a user really needs to override this config, they can always override it in the their project's `Kconfig` file.

Reply:
So that I should just completely remove the prompt here?

Reply:
> So that I should just completely remove the prompt here?

If you do that, then it cannot be set by `prj.conf` files, right?

Reply:
> So that I should just completely remove the prompt here?

Yes. I vote for that.

Reply:
> > So that I should just completely remove the prompt here?
> 
> If you do that, then it cannot be set by `prj.conf` files, right?

Correct. But I think you can still override it in `Kconfig` (file in project root).

Reply:
>On the other hand, I think that user needs possibility to configure size of this stack in some cases (e.g. to tailor the stack to save memory or to increase it on some architectures - we may not oversee everything and I would prefer not to make stack very big by default).

The user always has the option to redeclare the Kconfig option to override the value. The user can either define their Kconfig root or use the board's if they have their own board. Consensus here is to drop the prompt and let advanced users override it with redeclaration instead.

Reply:
> Consensus here is to drop the prompt and let advanced users override it with redeclaration instead.

Was just about to ask about that; for boolean values it's easy to just declare a new Kconfig and have that `select` the Kconfigs without a prompt, but that approach does not work for `int` values like this. 

So the solution is to redeclare it and override it?